src/ui/osx/TogglDesktop/Feature/Permission/SystemPermissionManager.swift

```diff
@@ -0,0 +1,169 @@
+
+                // Show alert to instruct the user to manually grant the permission
+                if isAlreadyRequestSystemPermission(permission) {
+                    presentScreenRecordingAlert {
+                        tryOpeningSystemPreference(for: permission)
+                    }
+                } else {
+                    // Trigger the system alert once
+                    triggerScreenRecordingPermissionAlert()
+                }
+            }
+        }
+    }
+}
+
+// MARK: Private
+
+extension SystemPermissionManager {
+
+    private func presentScreenRecordingAlert(_ complete: () -> Void) {
+        let alert = NSAlert()
+        alert.messageText = "Screen Recording permission not granted!"
+        alert.informativeText = "To get the Windows Name properly for the Timeline, TogglDesktop needs to be granted the Screen Recording permission in Security & Privacy in System Preferences .\n\nPlease open System Preferences -> Security & Privacy -> Privacy Tab -> Select Screen Recording and enable TogglDesktop app."
+        alert.alertStyle = .warning
+        alert.addButton(withTitle: "Open")
+        alert.addButton(withTitle: "Later")
+
+        let result = alert.runModal()
+        if result == .alertFirstButtonReturn {
+            complete()
+        }
+    }
+
+    private func isAlreadyRequestSystemPermission(_ permission: Permission) -> Bool {
+        switch permission {
+        case .screenRecording:
+            return UserDefaults.standard.bool(forKey: Constants.ScreenRecordingPermissionGranted)
+        }
+    }
+
+    private func tryOpeningSystemPreference(for permission: Permission) {
+
+        // Open the Screen Recording tab
+        // https://stackoverflow.com/a/48139877/3127477
+        if let url = URL(string: permission.urlScheme) {
+            NSWorkspace.shared.open(url)
+            return
+        }
+
+        // Old-fashioned way
+        // https://github.com/cho45/KeyCast/blob/master/KeyCast/Accessibility.swift
+        let url = URL(fileURLWithPath: "/System/Library/PreferencePanes/Security.prefPane")
+        NSWorkspace.shared.open(url)
+    }
+}
+
+// MARK: Screen Recording
+
+extension SystemPermissionManager {
+
+    // CGWindowListCopyWindowInfo is not reliable because some windows has the name even the permission is OFF
+    // Reliable way to check whether the permission is enable or not
+    // However it would trigger the system alert too
+    private func canRecordScreen() -> Bool {
+        let mainDisplay    = CGMainDisplayID()
+        let displayBounds  = CGDisplayBounds(mainDisplay)
+        let recordingQueue = DispatchQueue.global(qos: .background)
+        let displayStreamProps : [CFString : Any] = [
+            CGDisplayStream.preserveAspectRatio: kCFBooleanTrue,
+            CGDisplayStream.showCursor:          kCFBooleanTrue,
+            CGDisplayStream.minimumFrameTime:    60,
+        ]
+
+        let displayStream = CGDisplayStream(
```

Comment:
Since there is no official API to determine if the Screen Recording permission is granted or not -> We have to try creating the `CGDisplayStream` instance.
- If it is not `nill` -> the permission is granted.